src/main/java/com/digirati/taxonomy/manager/lookup/normalisation/TextNormaliser.java

```diff
@@ -0,0 +1,125 @@
+package com.digirati.taxonomy.manager.lookup.normalisation;
+
+import com.digirati.taxonomy.manager.lookup.model.Word;
+import com.google.common.annotations.VisibleForTesting;
+import edu.stanford.nlp.ling.CoreLabel;
+import edu.stanford.nlp.pipeline.CoreDocument;
+import edu.stanford.nlp.pipeline.StanfordCoreNLP;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Provides a mechanism for normalising any text inputs, including the terms to search for and the
+ * text in which to search.
+ */
+public class TextNormaliser {
+
+    private final Set<String> stopwords;
+
+    private final StanfordCoreNLP pipeline;
+
+    private final Function<String, CoreDocument> coreDocumentGenerator;
+
+    /**
+     * Initialises a {@link TextNormaliser} for normalising any input language. Currently, full
+     * lemmatisation is only supported for English due to a restriction of the NLP library that
+     * we're using, but language-specific tokenisation should still work
+     *
+     * @param stopwords a collection of common boilerplate words to remove from the text (e.g.
+     *     "the", "a", etc.)
+     * @param languageKey the ISO 639-1 language code for the text to be normalised
+     * @param languageName the name of the language of the text to be normalised
+     */
+    public static Future<TextNormaliser> initialiseNormaliser(
+            Set<String> stopwords, String languageKey, String languageName) {
+        return Executors.newSingleThreadExecutor()
```

Comment:
I'm not sure I see the point of submitting this task to an executor and returning a `Future`. Obviously the initialisation of the normaliser is a potentially expensive process, and could take some time to complete, but I don't see that alone as being a valid case for using multi-threading.

If the caller gets the `Future<TextNormaliser>` when they call "initialiseNormaliser` or `initialiseEnglishNormaliser`, what are they going to do with it? Answer: they're probably just going to poll it until it returns. Why not return `TextNormaliser` directly and let the call block until it's ready?

Even if the argument is made that the initialisation of the normaliser should happen in another thread, then that the caller should handle the multi-threading. It's not the responsibility of `TextNormaliser` to make assumptions about how the caller wants to handle long running / blocking method calls.

Reply:
This was in response to one of Gary's comments on the previous change - https://github.com/digirati-co-uk/digirati-taxonomy-manager/pull/6#discussion_r291534298

I don't have a preference either way as long as it's clear to the calling code that this is a long blocking call